### PR TITLE
feat: BE docker deployment (PLT-6855)

### PR DESCRIPTION
**Pull request by bogdan-manole**

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -12,7 +12,99 @@
...
+        script="cp ${pkgs.cacert}/etc/ssl/certs/ca-bundle.crt /etc/ssl/certs/ca-certificates.crt"
+        
+        #TODO: change this --unsafe-bypass-subscription-validation
+        args="$args --unsafe-bypass-subscription-validation"
+
+        echo $script >&2
+        eval "$script"
+
+        script="${flake.packages."plutus-certification:exe:plutus-certification"}/bin/plutus-certification $args"
+        echo $script >&2
```
Comment (demoiog) [inline flake.nix:75]: same here

File: flake.nix [hunk 2]
```diff
@@ -12,7 +12,99 @@
...
+        mkdir -p /tmp
+
+        # copy the certificate bundle to the right place
+        mkdir -p /etc/ssl/certs
+        script="cp ${pkgs.cacert}/etc/ssl/certs/ca-bundle.crt /etc/ssl/certs/ca-certificates.crt"
+        
+        #TODO: change this --unsafe-bypass-subscription-validation
+        args="$args --unsafe-bypass-subscription-validation"
+
+        echo $script >&2
```
Comment (demoiog) [inline flake.nix:71]: remove this after production due to secrets exposure in the logs

File: flake.nix [hunk 3]
```diff
@@ -12,7 +12,99 @@
...
+  
+    nixImage = pkgs.dockerTools.pullImage {
+      imageName = "nixos/nix";
+      imageDigest = "sha256:31b808456afccc2a419507ea112e152cf27e9bd2527517b0b6ca8639cc423501";
+      sha256 = "0bbw3r0civlcm3inj23fq8f25aw63rnaay09qjbrvfjd7pcfbyqn";
+      finalImageName = "nixos/nix";
+      finalImageTag = "2.15.0";
+    };
+    genFlake = flake.packages."dapps-certification-helpers:exe:generate-flake";
+    buildFlake = flake.packages."dapps-certification-helpers:exe:build-flake";
```
Comment (demoiog) [inline flake.nix:87]: this can be removed as well in the future

File: flake.nix [hunk 4]
```diff
@@ -12,7 +12,99 @@
...
+        paths = [ pkgs.curl pkgs.zsh pkgs.coreutils pkgs.nmon pkgs.cacert genFlake buildFlake ];
+        pathsToLink = [ "/bin" ];
+      };
+       runAsRoot = ''
+        rm -rf /etc/nix/nix.conf
+        echo "${nixConfig}" > /etc/nix/nix.conf
+       '';
+
+      config = {
+        Cmd = ["${builtins.trace entryPoint entryPoint}"];
```
Comment (demoiog) [inline flake.nix:104]: remove the trace

File: flake.nix [hunk 5]
```diff
@@ -29,12 +121,81 @@
...
+              export PORT=9671
+            fi
+            docker_args="$docker_args -p $PORT:$PORT"
+            
+            script="docker run --rm $docker_args ${imgAttributes.name}:${imgAttributes.tag}"
+            echo $script >&2
+            eval "$script"
+        '').outPath;
+      };
+      pushDockerImage = {
```
Comment (demoiog) [inline flake.nix:170]: add some readme for this

**Review by RSoulatIOHK — APPROVED** [review]
Looks good.

**Pull request merged**